color.go

```diff
@@ -0,0 +1,17 @@
+package lol
+
+// colorer is an interface that can be used to colorize an io.Writer. Each
+// colorer is designed to support a specific terminal color schema (256, 0, etc).
+type colorer interface {
+
+	// Rainbow will calculate the current color spectrum for RGB colors only.
+	// Each implementation other than truecolor will need to implement rainbow
+	// and handle it accordingly.
+	rainbow(freq float64, i float64)
```

Comment:
If we go with the interface approach (as defined here) we might want to pull rainbow() out.. it's very redundant